### PR TITLE
modules/image/repart: Fix stripNixStorePrefix

### PR DESCRIPTION
**Pull request by arianvp**
## Description of changes

It seems it wasn't working

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by arianvp** [comment]
With this change it still doesn't work. I get the following error now:

```
error: builder for '/nix/store/w8aa79xfx9ag1rlfmnd9ngaagxkc1npn-nixos.drv' failed with exit code 1;
       last 10 log lines:
       > /build/amended-repart.d/nixStore.conf:516: CopyFiles= path is not absolute, ignoring: zc9ryh74m3cyzzlkkm0pxsx0jq8jiqzl-tzdata-2023c
       > /build/amended-repart.d/nixStore.conf:517: CopyFiles= path is not absolute, ignoring: zg1dmivn1fhc0n7v3ra9yr6r0d32604s-X-Restart-Triggers
       > /build/amended-repart.d/nixStore.conf:518: CopyFiles= path is not absolute, ignoring: ziyf74lw0im3cpm8j7g9d0rswjk3y1nr-npth-1.6
       > /build/amended-repart.d/nixStore.conf:519: CopyFiles= path is not absolute, ignoring: zj1ah6hzvgw18fnp1xkx5w1wwjxgi7if-aws-sdk-cpp-1.11.118
       > /build/amended-repart.d/nixStore.conf:520: CopyFiles= path is not absolute, ignoring: zl4hzvv5rp9n0hr0xgf0kxl5vzwyzp33-libcap-2.69
       > /build/amended-repart.d/nixStore.conf:521: CopyFiles= path is not absolute, ignoring: zm9jq42h6kqfbxb9mh3gk2p1x4x4s60i-mdadm-4.2
       > /build/amended-repart.d/nixStore.conf:522: CopyFiles= path is not absolute, ignoring: zmxrj3pgfqc38jwycgdyaqs4nggm4pxd-libressl-3.7.3-man
       > /build/amended-repart.d/nixStore.conf:523: CopyFiles= path is not absolute, ignoring: zphbsjgw5xw9lwkgl1rvh8p79jd3z530-fc-53-no-bitmaps.conf
       > /build/amended-repart.d/nixStore.conf:524: CopyFiles= path is not absolute, ignoring: zr6pvi1b1cn5skj31wy7hi3yvfz469i9-unit-systemd-network-wait-online-.service
       > /build/amended-repart.d/nixStore.conf:1: Cannot format erofs filesystem without source files, refusing
       For full logs, run 'nix log /nix/store/w8aa79xfx9ag1rlfmnd9ngaagxkc1npn-nixos.drv'.

```

**Comment by arianvp** [comment]
should work now. Maybe we should add a NixOS test for this?

File: nixos/modules/image/amend-repart-definitions.py [hunk 1]
```diff
@@ -53,7 +53,7 @@ def add_closure_to_definition(
 
             source = Path(line.strip())
             target = str(source.relative_to("/nix/store/"))
-            target = f":{target}" if strip_nix_store_prefix else ""
+            target = f":/{target}" if strip_nix_store_prefix else ""
```
Comment (nikstur) [inline nixos/modules/image/amend-repart-definitions.py:56]: Are you sure this is correct? Maybe we can add a test that actually does the stripping.

Comment (arianvp) [reply nixos/modules/image/amend-repart-definitions.py:56]: Without this I get:

```
 > /build/amended-repart.d/nixStore.conf:516: CopyFiles= path is not absolute, ignoring: zc9ryh74m3cyzzlkkm0pxsx0jq8jiqzl-tzdata-2023c
       > /build/amended-repart.d/nixStore.conf:517: CopyFiles= path is not absolute, ignoring: zg1dmivn1fhc0n7v3ra9yr6r0d32604s-X-Restart-Triggers
       > /build/amended-repart.d/nixStore.conf:518: CopyFiles= path is not absolute, ignoring: ziyf74lw0im3cpm8j7g9d0rswjk3y1nr-npth-1.6
       > /build/amended-repart.d/nixStore.conf:519: CopyFiles= path is not absolute, ignoring: zj1ah6hzvgw18fnp1xkx5w1wwjxgi7if-aws-sdk-cpp-1.11.118
       > /build/amended-repart.d/nixStore.conf:520: CopyFiles= path is not absolute, ignoring: zl4hzvv5rp9n0hr0xgf0kxl5vzwyzp33-libcap-2.69
       > /build/amended-repart.d/nixStore.conf:521: CopyFiles= path is not absolute, ignoring: zm9jq42h6kqfbxb9mh3gk2p1x4x4s60i-mdadm-4.2
       > /build/amended-repart.d/nixStore.conf:522: CopyFiles= path is not absolute, ignoring: zmxrj3pgfqc38jwycgdyaqs4nggm4pxd-libressl-3.7.3-man
       > /build/amended-repart.d/nixStore.conf:523: CopyFiles= path is not absolute, ignoring: zphbsjgw5xw9lwkgl1rvh8p79jd3z530-fc-53-no-bitmaps.conf
       > /build/amended-repart.d/nixStore.conf:524: CopyFiles= path is not absolute, ignoring: zr6pvi1b1cn5skj31wy7hi3yvfz469i9-unit-systemd-network-wait-online-.service
```

**Comment by nikstur** [comment]
@ofborg test appliance-repart-image

**Review by nikstur — APPROVED** [review]
Looks good thank you!

**Pull request merged**